File: couch-server/src/main/java/com/fronde/server/services/setting/SettingCriteria.java
package com.fronde.server.services.setting;

import com.fronde.server.domain.criteria.AbstractCriteria;

public class SettingCriteria extends AbstractCriteria {

  private String name;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

}

File: angular-client/src/app/setting/setting.criteria.ts
import { AbstractCriteria } from '../domain/criteria/abstract-criteria';
import { Params } from '@angular/router';

export class SettingCriteria extends AbstractCriteria {
  name: string;

  populateFromParams(params: Params) {
    this.doPopulateFromParams(params, [], ['pageSize', 'pageNumber'], [], []);
  }

  toParams() {
    return this.doToParams(this);
  }
}
